opcua/address_space.py

```diff
@@ -395,13 +395,17 @@ def set_attribute_value(self, nodeid, attr, value):
             attval = node.attributes[attr]
             old = attval.value
             attval.value = value
+            cbs = []
             if old.Value != value.Value:  # only send call callback when a value change has happend
-                for k, v in attval.datachange_callbacks.items():
-                    try:
-                        v(k, value)
-                    except Exception as ex:
-                        self.logger.exception("Error calling datachange callback %s, %s, %s", k, v, ex)
-            return ua.StatusCode()
+                cbs = list(attval.datachange_callbacks.items())
```

Comment:
It does not matter, list() is called with lock. I use it to do a shadow copy. items() doesn't return a list but a view in Python3.